plugin/encryption_udf/encryption_udf.cc

```diff
@@ -0,0 +1,258 @@
+#include <stdexcept>
+#include <string>
+
+#include <mysql/plugin.h>
+
+#include <mysqlpp/udf_wrappers.hpp>
+
+#include <opensslpp/core_error.hpp>
+#include <opensslpp/rsa_key.hpp>
+#include <opensslpp/rsa_padding.hpp>
+
+namespace {
+
+//
+// CREATE_ASYMMETRIC_PRIV_KEY(@algorithm, {@key_len|@dh_secret})
+// This functions generates a private key using the given algorithm
+// (@algorithm) and key length (@key_len) or Diffie-Hellman secret(@dh_secret),
+// and returns the key as a binary string in PEM format.
+// If key generation fails, the result is NULL.
+//
+class create_asymmetric_priv_key_impl {
+ public:
+  create_asymmetric_priv_key_impl(mysqlpp::udf_context &ctx) {
+    DBUG_TRACE;
+
+    if (ctx.get_number_of_args() != 2)
+      throw std::invalid_argument("Function requires exactly two arguments");
+
+    // result
+    ctx.mark_result_const(false);
+    ctx.mark_result_nullable(true);
+
+    // arg0 - @algorithm
+    ctx.mark_arg_nullable(0, false);
+    ctx.set_arg_type(0, STRING_RESULT);
+
+    // arg1 - @key_len
+    ctx.mark_arg_nullable(1, false);
+    ctx.set_arg_type(1, INT_RESULT);
+  }
+  ~create_asymmetric_priv_key_impl() { DBUG_TRACE; }
+
+  mysqlpp::udf_result_t<STRING_RESULT> calculate(
+      const mysqlpp::udf_context &args);
```

Comment:
Parameter name in declaration and implementation differs

Reply:
Fixed